src/main/java/edu/harvard/iq/dataverse/DOIDataCiteServiceBean.java

```diff
@@ -171,33 +178,79 @@ public void deleteRecordFromCache(Dataset datasetIn){
         }
     }
 
+    /*
+     * Deletes a DOI if it is in DRAFT/RESERVED state or removes metadata and changes it from PUBLIC/FINDABLE to REGISTERED.
+     */
+    @Override
+    public void deleteIdentifier(DvObject dvObject) throws Exception {
+        logger.log(Level.FINE,"deleteIdentifier");
+        String identifier = getIdentifier(dvObject);
+        Map<String, String> doiMetadata = new HashMap<>();
+        try {
+            doiMetadata = doiDataCiteRegisterService.getMetadata(identifier);
+        } catch (Exception e) {
+            logger.log(Level.WARNING, "deleteIdentifier: get matadata failed. " + e.getMessage(), e);
+        }
+
+        String idStatus = doiMetadata.get("_status");
+        if ( idStatus != null ) {
+            switch ( idStatus ) {
+                case RESERVED:
+                case DRAFT:    
+                    logger.log(Level.INFO, "Delete status is reserved..");
+                    try {
+                    	//service only removes the identifier from the cache (since it was written before DOIs could be registered in draft state)
+                        doiDataCiteRegisterService.deleteIdentifier(identifier);
+                        //So we call the deleteDraftIdentifier method below until things are refactored
+                        deleteDraftIdentifier(dvObject);
+                    } catch (Exception e) {
+                        logger.log(Level.WARNING, "delete failed: " + e.getMessage(), e);
+                    }
+                    break;
+
+                case PUBLIC:
+                case FINDABLE:
+                    //if public then it has been released set to unavailable and reset target to n2t url
+                    Map<String, String> metadata = addDOIMetadataForDestroyedDataset(dvObject);
+                    metadata.put("_status", "registered");
+                    metadata.put("_target", getTargetUrl(dvObject));                   
+                    doiDataCiteRegisterService.deactivateIdentifier(identifier, metadata, dvObject);
+                    break;
+            }
+        }
+    }
+        
     /**
      * Deletes DOI from the DataCite side, if possible. Only "draft" DOIs can be
      * deleted.
      */
-    @Override
-    public void deleteIdentifier(DvObject dvObject) throws IOException {
+    private void deleteDraftIdentifier(DvObject dvObject) throws IOException {
```

Comment:
What happens if this deleteDraftIdentifier method is called on a non-DataCite installation (Handle or EZID)?

Reply:
It's private in the DataCite code. THe code here could be put in the public deleteIdentifier method - just decided to keep it separate since it will hopefully become part of the doiDataCiteRegisterService.deleteIdentifier(identifier); call on line 204 at some point.

Reply:
Ah, right, I forgot we're in the DataCite-specific bean. Thanks.